ui/app/src/views/ViewDashboard.tsx

```diff
@@ -11,32 +11,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { useQueryParams, StringParam } from 'use-query-params';
-import { DashboardResource } from '@perses-dev/core';
 import { ViewDashboard as DashboardView } from '@perses-dev/dashboards';
+import { useParams } from 'react-router-dom';
 import { useDatasourceApi } from '../model/datasource-api';
-import { useSampleData } from '../utils/temp-sample-data';
+import { useDashboard } from '../model/dashboard-client';
 
-const DEFAULT_DASHBOARD_ID = 'node-exporter-full';
+interface ViewSingleDashboardProperty {
+  dashboardName: string;
+  projectName: string;
+}
+
+function ViewSingleDashboard(props: ViewSingleDashboardProperty) {
```

Comment:
not the best name for a component. It would be cool to find a better one

Reply:
Feels like this _shouldn't_ (in theory) need another component in here? Are we basically just doing this so that we can show `ViewDashboard` as the home page right now (i.e. not have a project ID and dashboard ID)?

Reply:
I actually create another component, otherwise the linter were complaining I'm creating a hook after a potential return.

In `ViewDashboard` I'm first getting the project and dashboard name. But since these value can be undefined I had to check first.

Maybe I can create a dummy blank page and so `ViewDashboard` is not used as the home page anymore. And then I could just throw an error as you suggested

Reply:
I removed the component `ViewSingleDashboard` and throw an error when the dashboard and project name are undefined. It's better like that indeed :). Thanks !